refactor(language_server):  move restart to `Tool::handle_watched_file_change`  trait method

Sysix opened the pull request:
Returning `ToolRestartChanges` like `handle_configuration_change` does.
The logic in `WorkspaceWorker/Backend` can be shared later for simplified usage.

camc314 commented:
### Merge activity

* **Nov 9, 12:21 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 9, 12:21 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Nov 9, 12:26 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(language_server):  move restart to `Tool::handle_watched_file_change`  trait method (#15491)

Returning `ToolRestartChanges` like `handle_configuration_change` does.
The logic in `WorkspaceWorker/Backend` can be shared later for simplified usage.
